ArmPkg/Drivers/MmCommunicationDxe/MmCommunication.c

```diff
@@ -259,8 +259,17 @@ GetMmCompatibility (
 
   MmVersion = MmVersionArgs.Arg0;
 
+  // MU_CHANGE - CodeQL Change - unsigned-comparison-zero
+  // Make MM_CALLER_MINOR_VER more restrictive. Rquire
+  //  exact minor version match rather than >=.
+  // Add static_assert to check if/when verison change
+  // occurs.
+  STATIC_ASSERT (
+    MM_CALLER_MINOR_VER == 0,
+    "MM_CALLER_MINOR_VERSION has changed!"

```

Comment:
Is this a new requirement that the minor version cannot be zero?

Reply:
If so, it should be marked as a breaking change.

Reply:
This stems from the CodQl error that is was comparing 0 (MM_CALLER_MINOR_VERSION) against greater than or equal to zero, which was a usless comparison.

The minor version has not changed in 6 years, since it was introduced.

Additionally, the >= comparison meant they were trying to make it future minor version compatible, meaning that a minor version update would not cause a break in the functionality.

After discussing with kuqin12, it was decided to make the check more restrictive (meaning it needs to match the version) and to add a static assert to break on a minor version change instead of just showing an error in the debug log.

I can mark this as breaking, but I assume that if it has not changed in 6 years, the chance of it being updated is minimal.

Reply:
I misinterpreted the assert logic earlier. A minor version typically exists to allow version tracking with breaking compatibility, so this seems a little strange in that regard.

Also, the comparison is only considered useless because of the current value of `MM_CALLER_MINOR_VER` and the fact that it's a static value. The original intention is still valid though the context is not apparent to CodeQL.

Is the expectation in the current implementation that the `STATIC_ASSERT` condition is updated over time? If it stays, I think it needs to explain in more detail what the impact and resolution is.

Reply:
I updated the comment around the added STATIC_ASSERT to say that it should be updated over time if/when the caller version ever changes.